backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs

```diff
@@ -87,6 +87,15 @@ public async Task<ICollection<LessonDto>> GetAllLessonsAsync(FiltersRequest filt
         return dayCards;
     }
 
+    private DateTime GetMondayDate(DateTime date)
+    {
+        if(date.DayOfWeek == DayOfWeek.Sunday)
+        {
+            return date.AddDays(6).Date;
+        }
+        return date.AddDays(-((date.DayOfWeek - DayOfWeek.Monday) % 7)).Date;
+    }
```

Comment:
I guess something is wrong)
1. Sunday is 6
2. All other days are -5...0

And do you still need %7?
+ I will move minus to GetDayCardsOfWeekAsync function